When we compile Perl 5 blead with g++10 as our C-compiler, we get two new build-time warnings in comparison with g++9.

```
$ ~/bin/perl/parse-one-warnings-group 6128f436ce.freebsd.g++10.threaded.maketp.output.txt.gz Wparentheses
File:  6128f436ce.freebsd.g++10.threaded.maketp.output.txt.gz

[
  {
    char   => 72,
    group  => "Wparentheses",
    line   => 3293,
    source => "gv.c",
    text   => "suggest parentheses around assignment used as truth value",
  },
  {
    char   => 73,
    group  => "Wparentheses",
    line   => 3300,
    source => "gv.c",
    text   => "suggest parentheses around assignment used as truth value",
  },
]
```
Excerpt from `make test_prep` output:
```
g++10 -c -DPERL_CORE -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -ansi -O2 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wwrite-strings gv.c
gv.c: In function 'SV* Perl_amagic_call(PerlInterpreter*, SV*, SV*, int, int)':
gv.c:3293:72: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
 3293 |                || ((cv = cvp[off = add_amg]) && (force_cpy = 0, postpr = 1))) {
      |                                                                 ~~~~~~~^~~
gv.c:3300:73: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
 3300 |                || ((cv = cvp[off = subtr_amg]) && (force_cpy = 0, postpr=1))) {
      |                                                                   ~~~~~~^~
```
We can silence these warnings with the small changes in this pull request.  The evidence from [smoke-testing](http://perl.develop-help.com/?b=smoke-me%2Fjkeenan%2Fg%2B%2B10-build-time-warnings-20210913) and from [CI](https://github.com/Perl/perl5/actions/runs/1231631961) suggest that these changes do no harm elsewhere.

Please review.

Thank you very much.
Jim Keenan

--- a/gv.c
+++ b/gv.c
@@ -3290,14 +3290,16 @@ Perl_amagic_call(pTHX_ SV *left, SV *right, int method, int flags)
          case inc_amg:
            force_cpy = 1;
            if ((cv = cvp[off=add_ass_amg])
-               || ((cv = cvp[off = add_amg]) && (force_cpy = 0, (postpr = 1)))) {
+               || ((cv = cvp[off = add_amg])
+                   && (force_cpy = 0, (postpr = 1)))) {
              right = &PL_sv_yes; lr = -1; assign = 1;
            }
            break;
          case dec_amg:
            force_cpy = 1;
            if ((cv = cvp[off = subtr_ass_amg])
-               || ((cv = cvp[off = subtr_amg]) && (force_cpy = 0, (postpr=1)))) {
+               || ((cv = cvp[off = subtr_amg])
+                   && (force_cpy = 0, (postpr=1)))) {
              right = &PL_sv_yes; lr = -1; assign = 1;
            }
            break;

--- a/gv.c
+++ b/gv.c
@@ -3290,14 +3290,14 @@ Perl_amagic_call(pTHX_ SV *left, SV *right, int method, int flags)
          case inc_amg:
            force_cpy = 1;
            if ((cv = cvp[off=add_ass_amg])
-               || ((cv = cvp[off = add_amg]) && (force_cpy = 0, postpr = 1))) {
+               || ((cv = cvp[off = add_amg]) && (force_cpy = 0, (postpr = 1)))) {
              right = &PL_sv_yes; lr = -1; assign = 1;
            }
            break;
          case dec_amg:
            force_cpy = 1;
            if ((cv = cvp[off = subtr_ass_amg])
-               || ((cv = cvp[off = subtr_amg]) && (force_cpy = 0, postpr=1))) {
+               || ((cv = cvp[off = subtr_amg]) && (force_cpy = 0, (postpr=1)))) {
              right = &PL_sv_yes; lr = -1; assign = 1;
            }
            break;

